pkg/threescale/backend/backend.go

```diff
@@ -148,15 +144,18 @@ func (b *Backend) isAuthorized(application *Application, affectedMetrics api.Met
 
 out:
 	for metric, incrementBy := range affectedMetrics {
-		cachedValue, ok := application.LimitCounter[metric]
-		if ok {
-			for _, limit := range cachedValue {
-				if limit.current+incrementBy > limit.max {
-					authorized = false
-					break out
-				}
-			}
+		cachedValue, ok := application.LocalState[metric]
+
+		if !ok || len(cachedValue) < 1 {
+			continue
 		}
+
+		lowestKnownGranularity := cachedValue[0]
```

Comment:
yes, had added a comment to the Application stating that this is a requirement for someone caching an `Application` and in this case, the person(s) caching it are us so felt it would be implied but happy to call it out here